R/visualize.R

```diff
@@ -0,0 +1,237 @@
+#' * If provided a ([PredictionClassif]). Then only one point will indicate the accuracy and fairness metrics for the current predictions.
+#' * If provided a ([ResampleResult]). Then the plot will compare the accuracy and fairness metrics for the same model, but different resampling iterations.
+#' * If provided a ([BenchmarkResult]). Then the plot will compare the accuracy and fairness metrics for all models and all resampling iterations.
+#'
+#' @param ...
+#' Arguments to be passed to methods. Such as:
+#'
+#' * `fairness_measure` ([Measure])\cr
+#' The fairness measures that will evaluated on object, could be single measure (`msr`) or list of measures (`msrs`), check [mlr3::Measure]. Default measure set to be \code{msr("fairness.fpr")}
+#'
+#' * `accuracy_measure` ([Measure])\cr
+#' The accuracy measure that will evaluated on object, could be single measure \link{msr} or measures \link{msrs}. Default measure set to be \code{\link[mlr3:MeasureClassif]{msr("classif.acc")}}
+#'
+#' * `task` ([TaskClassif])\cr
+#' The data task that contains the protected column, only required when the class of object is ([PredictionClassif])
+#'
+#'
+#'
+#' @examples
+#' library(mlr3fairness)
+#' library(mlr3learners)
+#' library(mlr3)
+#' library(ggplot2)
+#' library(data.table)
+#'
+#' # Setup the Fairness Measures and tasks
+#' task = tsk("adult_train")
+#' learner = lrn("classif.ranger", predict_type = "prob")
+#' learner$train(task)
+#' predictions = learner$predict(task)
+#' design = benchmark_grid(
+#'   tasks = tsk("adult_train"),
+#'   learners = lrns(c("classif.ranger", "classif.rpart"),
+#'                  predict_type = "prob", predict_sets = c("train", "test")),
+#'   resamplings = rsmps("cv", folds = 3)
+#' )
+#'
+#' bmr = benchmark(design)
+#' fairness_measure = msr("fairness.tpr")
+#' fairness_measures = msrs(c("fairness.tpr", "fairness.fnr"))
+#'
+#' fairness_accuracy_tradeoff(predictions, fairness_measure, task)
+#' fairness_accuracy_tradeoff(bmr, fairness_measure)
+#' @export
+fairness_accuracy_tradeoff <- function(object, ...){
+  UseMethod("fairness_accuracy_tradeoff")
+}
+
+#' @export
+fairness_accuracy_tradeoff.PredictionClassif <- function(object, fairness_measure = msr("fairness.fpr"), task, acc_measure = msr("classif.acc"), ...){
+  data = data.table(accuracy = object$score(acc_measure),
+                    fairness = object$score(fairness_measure, task))
+  ggplot(data, aes(x = accuracy, y=fairness)) + geom_point()
+}
+
+#' @export
+fairness_accuracy_tradeoff.BenchmarkResult <- function(object, fairness_measure = msr("fairness.fpr"), acc_measure = msr("classif.acc"), ...){
+  data = data.table(model = object$score(fairness_measure)[, "learner_id"],
+                    accuracy = object$score(acc_measure)[, acc_measure$id, with = F],
+                    metrics = object$score(fairness_measure)[, fairness_measure$id, with = F])
+  colnames(data) <- c("model", "accuracy", "metrics")
```

Comment:
this should already be ensured above?

Reply:
Yeap, indeed. Sorry.

Reply:
Better to extract the column vector then, i.e. `object$score(fairness_measure)[, "learner_id"][[1]]`.

Reply:
Done